### PR TITLE
docs(madr): filter metrics

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
MADR about filter metrics.

xref #1755

### CODE REVIEW

**Comment by lobkovilya** [comment]
Maybe we can drastically reduce the number of metrics by setting `/stats?usedonly` by default. It seems like it can work together with `filter`

**Review by lobkovilya — APPROVED** [review]

File: docs/madr/decisions/004-metric-filter.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+Envoy can receive one [filter with regex](https://www.envoyproxy.io/docs/envoy/latest/operations/admin#get--stats?filter=regex)
+_Note:_ filter works on statsd names, not prometheus names, when configuring filter you need to take that into account. 
+You can only provide one filter, so `envoyFilterRegex` cannot be a list.
+
+Then, in prometheus_endpoint_generator.go we can configure route to add this query param, so the `:5670` is still `/metrics`, but it adds the filter under the hood.
+Then, metrics hijacker receives a request with query param and there are two options:
+* It passes it only to Envoy.
+* It passes it to Envoy and all the apps specified in aggregate.
+I think the first option is safer.
```
Comment (lahabana) [inline docs/madr/decisions/004-metric-filter.md:58]: Agreed I think the 2nd option is a different feature and we'd want to have a filter per "aggregate"

**Review by lahabana — APPROVED** [review]

File: docs/madr/decisions/004-metric-filter.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+* Filter metrics dynamically on scrape request
+* Filter metrics statically on Envoy bootstrap
+* Customizable bootstrap
+
+## Decision Outcome
+
+Chosen option:
+"Filter metrics dynamically on scrape request", because it follows Kuma philosophy of being as dynamic as possible.
+Additionally, as a followup we should implement "Customizable bootstrap", because it both enables static metrics and other advanced use cases.
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply docs/madr/decisions/004-metric-filter.md]: added

**Comment by jakubdyszkiewicz** [comment]
Good suggestion @lobkovilya 
Changed format to set the place for usedOnly.

**Review by lukidzi — APPROVED** [review]

**Pull request merged**